pkgs/stdenv/generic/check-meta.nix

```diff
@@ -278,6 +421,8 @@ let
     unfree = bool;
     unsupported = bool;
     insecure = bool;
+    # This is checked in more detail further down
+    problems = attrsOf attrs;
```

Comment:
Don't use `types.attrs`, it has poor merge semantics. You can use `types.attrsOf types.raw` or `types.attrsOf types.anything` (see the type docs for the difference), or even better, specify the full type using submodules.

Reply:
We are not using the module system so there is never going to be any merging, but fixed anyway. Specifying the full type using submodules does not help because we are not really using the module system.